### DISCUSSION TITLE
Prisma (docker): "Server selection timeout: No available servers." (with repo example)

### DISCUSSION THREAD

**Discussion by dejurin** [prisma/prisma#21399] [Q&A] [answered]
### Question

Hello, I can't connect **prisma** to **mongodb** docker container.

I got this message:

```
------
 > [app build 5/7] RUN pnpm dlx prisma db pull --force:
1.745 Environment variables loaded from .env
1.748 Datasource "db": MongoDB database "my_db" at "mongodb:27017"
1.750 
1.752 - Introspecting based on datasource defined in prisma/schema.prisma
31.78 Error: MongoDB error
31.78 Kind: Server selection timeout: No available servers. Topology: { Type: Unknown, Servers: [ { Address: mongodb:27017, Type: Unknown, Error: Kind: I/O error: failed to lookup address information: Name does not resolve, labels: {} } ] }, labels: {}
31.78 
31.78 
31.78 ✖ Introspecting based on datasource defined in prisma/schema.prisma
31.78 
------
```
full log: https://github.com/dejurin/mongodb-prisma-docker-example/blob/main/log.txt

### How to reproduce (optional)

I made repo with my example: https://github.com/dejurin/mongodb-prisma-docker-example

In this repo I include 2 cases: **app** - prisma (TS/JS), and **pyapp** - pymongo to make sure the problem is not with mongodb.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: macOS 14.0 (Apple M1 Max)
- Database: MongoDB 7
- Node.js version: v20.8.0
- Docker version: 4.24.0

- Run `prisma -v` to see your Prisma version and paste it
```
#12 2.786 prisma                  : 5.4.1
#12 2.786 @prisma/client          : 5.4.1
#12 2.786 Current platform        : linux-musl-arm64-openssl-3.0.x
#12 2.786 Query Engine (Node-API) : libquery-engine 2f302df92bd8945e20ad4595a73def5b96afa54f (at ../../pnpm/store/v3/tmp/dlx-1/node_modules/.pnpm/@prisma+engines@5.4.1/node_modules/@prisma/engines/libquery_engine-linux-musl-arm64-openssl-3.0.x.so.node)
#12 2.786 Schema Engine           : schema-engine-cli 2f302df92bd8945e20ad4595a73def5b96afa54f (at ../../pnpm/store/v3/tmp/dlx-1/node_modules/.pnpm/@prisma+engines@5.4.1/node_modules/@prisma/engines/schema-engine-linux-musl-arm64-openssl-3.0.x)
#12 2.786 Schema Wasm             : @prisma/prisma-schema-wasm 5.3.1-2.61e140623197a131c2a6189271ffee05a7aa9a59
#12 2.786 Default Engines Hash    : 2f302df92bd8945e20ad4595a73def5b96afa54f
#12 2.786 Studio                  : 0.494.0
```

**Comment by dejurin** [comment, marked as answer]
I updated if you remove (Dockerfile) the 
- `pnpm dlx prisma init --datasource-provider mongodb `
and
- `pnpm dlx prisma db pull --force`
then the prism connects and works fine.

So I conclude that initialization and schema creation should be done on the local computer before create docker compose. In my case, I tried to do all the steps while creating the containers.

Old:
```
RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --frozen-lockfile && \
    pnpm dlx prisma init --datasource-provider mongodb && \
    pnpm dlx prisma db pull --force && \
    pnpm dlx prisma generate && \
    pnpm run build
```
New
```
RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --frozen-lockfile && \
    pnpm dlx prisma generate && \
    pnpm run build
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
